xcode/Subconscious/Shared/Components/Common/SubtextTextViewRepresentable.swift

```diff
@@ -402,10 +324,30 @@ struct SubtextTextViewRepresentable: UIViewRepresentable {
             context.coordinator.isUIViewUpdating = false
         }
 
-        if (view.text != state.text) {
+        // When a key is pressed, updateUIView is called multiple times. The first time, state has
+        // not yet been updated to hold the latest text BUT view.text has already updated.
+        // This can result in thrashing the TextView and killing performance.
+        
+        // Pathological scenario:
+        // state.text = "abc"
+        // view.text = "abc"
+        // > user types "d"
+        // view.text = "abcd"
+        // updateUIView() called
+        // we set view.text = state.text
+        // now, view.text = "abc"
+        // THEN state.text updates to "abcd"
+        // updateUIVIew() called again
+        // we set view.text = state.text
+        // view.text = "abcd"
+        
+        // so we set view.text TWICE when we could set it zero times
+        // caching the last known rendered value allows us to detect and sidestep this scenario
```

Comment:
I'm a bit unclear about how this dirty flag approach might play with bi-directional data flow (from user to state, from state to UITextView).

I think the right course of action may be to land this, and then think about ways we might refactor.

Reply:
We might approach this like we do the block editor:

- UIViewRepresentable gets its own `@ObservedObject` store
- All UITextView events are sent as actions to store, updating state
- UIViewRepresentable subscribes to store state publisher, driving renders via publisher instead of update method
- Check UIView before setting state to ensure they're actually different

This seems to side-step a number of issues in the block editor, including the purple warning, since it guarantees view updates are downstream of state updates, and that those view updates always happen on the main thread.